RHDHBUGS-232: Changes to RHDH 1.5 release notes (#1078)
redhat-developer/red-hat-developers-documentation-rhdh: modules/release-notes/ref-release-notes-breaking-changes.adoc

* link:https://issues.redhat.com/browse/RHIDP-5812[RHIDP-5812]

[id="breaking-change-rhidp-5568"]
== The `dynamicPlugins.cache.volumeClaimSpec` field is removed from the Helm Chart, dynamic plugin storage is now ephemeral by default

> The `dynamicPlugins.cache.volumeClaimSpec` field has been removed from `values.yaml`, making dynamic plugin storage ephemeral by default. If you want to enable plugin caching persistence, you must now manually configure a `PersistentVolumeClaim` and mount it using the `extraVolumes` and `extraVolumeMounts` fields in the Helm chart values.

Reviewer: Would it make sense to also link to the actual doc on how to configure this?
https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.5/html/configuring_red_hat_developer_hub/con-dynamic-plugin-cache_running-behind-a-proxy#creating-a-pvc-for-the-dynamic-plugin-cache-using-the-helm-chart

> The `dynamicPlugins.cache.volumeClaimSpec` field has been removed from `values.yaml`, making dynamic plugin storage ephemeral by default using `emptyDir: {}`. Persistent plugin caching remains fully supported, and you must now manually configure `PersistentVolumeClaim` and mount it using the `extraVolumes` and `extraVolumeMounts` fields in the Helm chart values.

Reviewer: ```suggestion
The `dynamicPlugins.cache.volumeClaimSpec` field has been removed from `values.yaml`, making dynamic plugin storage ephemeral by default. If you want to enable plugin caching persistence, you must now manually configure a `PersistentVolumeClaim` and mount it using the `extraVolumes` and `extraVolumeMounts` fields in the Helm chart values.
```

2 points here:
- The Helm Chart does not use `emptyDir` for the dynamic plugins storage. It is a [generic ephemeral volume](https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/#generic-ephemeral-volumes) claim (quite similar to `emptyDir` but not exactly the same): https://github.com/redhat-developer/rhdh-chart/blob/release-1.5/charts/backstage/values.yaml#L148-L156
- I feel like the second sentence was not clear for users to re-enable plugin caching.

Author: > 2 points here:
> 
> * The Helm Chart does not use `emptyDir` for the dynamic plugins storage. It is a [generic ephemeral volume](https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/#generic-ephemeral-volumes) claim (quite similar to `emptyDir` but not exactly the same): https://github.com/redhat-developer/rhdh-chart/blob/release-1.5/charts/backstage/values.yaml#L148-L156
> * I feel like the second sentence was not clear for users to re-enable plugin caching.

Thank you for these suggestions, @rm3l.
